src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -306,6 +306,8 @@ void c_typecheck_baset::typecheck_expr_main(exprt &expr)
     // replace declaration by symbol expression
     symbol_exprt bound=to_symbol_expr(expr.op0().op0());
     expr.op0().swap(bound);
+
+    implicit_typecast_bool(expr.op1());
```

Comment:
I'm aware that this was covered by existing tests where this might have been redundant, but what about cases where the expression was _not_ Boolean? I would expect some failing tests to be added.

Reply:
Added to an existing test!